src/codegen/updater.cpp

```diff
@@ -0,0 +1,173 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// updater.cpp
+//
+// Identification: src/codegen/updater.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/updater.h"
+#include "codegen/transaction_runtime.h"
+#include "common/container_tuple.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "executor/executor_context.h"
+#include "storage/data_table.h"
+#include "storage/tile_group_header.h"
+#include "storage/tile_group.h"
+#include "storage/tile.h"
+#include "storage/tuple.h"
+#include "type/abstract_pool.h"
+#include "type/types.h"
+#include "type/value.h"
+
+namespace peloton {
+namespace codegen {
+
+void Updater::Init(concurrency::Transaction *txn, storage::DataTable *table,
```

Comment:
Modify the initialization by removing `Transaction`, and add `ExecutorContext`. This way, you get both the memory pool and transaction handle at initialization time. You can then remove `GetPool()` and remove `ExecutorContext` from `Update()`